src/openrct2/park/ParkPreview.cpp

```diff
@@ -0,0 +1,233 @@
+{
+    static std::optional<PreviewImage> generatePreviewMap();
+    static std::optional<PreviewImage> generatePreviewScreenshot();
+
+    ParkPreview generatePreviewFromGameState(const GameState_t& gameState)
+    {
+        ParkPreview preview{
+            .parkName = gameState.Park.Name,
+            .parkRating = gameState.Park.Rating,
+            .year = gameState.Date.GetYear(),
+            .month = gameState.Date.GetMonth(),
+            .day = gameState.Date.GetDay(),
+            .parkUsesMoney = !(gameState.Park.Flags & PARK_FLAGS_NO_MONEY),
+            .cash = gameState.Cash,
+            .numRides = static_cast<uint16_t>(RideManager().size()),
+            .numGuests = static_cast<uint16_t>(gameState.NumGuestsInPark),
+        };
+
+        if (auto image = generatePreviewMap(); image != std::nullopt)
+            preview.images.push_back(*image);
+
+        if (auto image = generatePreviewScreenshot(); image != std::nullopt)
+            preview.images.push_back(*image);
+
+        return preview;
+    }
+
+    static PaletteIndex getPreviewColourByTilePos(const TileCoordsXY& pos)
+    {
+        PaletteIndex colour = PALETTE_INDEX_0;
+
+        auto tileElement = MapGetFirstElementAt(pos);
+        if (tileElement == nullptr)
+            return colour;
+
+        do
+        {
+            switch (tileElement->GetType())
+            {
+                case TileElementType::Surface:
+                {
+                    auto* surfaceElement = tileElement->AsSurface();
+                    if (surfaceElement == nullptr)
+                    {
+                        colour = PALETTE_INDEX_0;
+                        break;
+                    }
+
+                    if (surfaceElement->GetWaterHeight() > 0)
+                    {
+                        colour = PALETTE_INDEX_195;
+                        break;
+                    }
+
+                    const auto* surfaceObject = surfaceElement->GetSurfaceObject();
+                    if (surfaceObject != nullptr)
+                    {
+                        colour = surfaceObject->MapColours[1];
+                    }
+                    break;
+                }
+
+                case TileElementType::Path:
+                    colour = PALETTE_INDEX_17;
+                    break;
+
+                case TileElementType::Track:
+                    colour = PALETTE_INDEX_183;
+                    break;
+
+                case TileElementType::SmallScenery:
+                case TileElementType::LargeScenery:
+                    colour = PALETTE_INDEX_99; // 64
+                    break;
+
+                case TileElementType::Entrance:
+                    colour = PALETTE_INDEX_186;
+                    break;
+
+                default:
+                    break;
+            }
+        } while (!(tileElement++)->IsLastForTile());
+
+        return colour;
+    }
+
+    // 0x0046DB4C
+    static std::optional<PreviewImage> generatePreviewMap()
```

Comment:
I may have missed why, but I don’t think I have seen this preview map show up anywhere?

Reply:
You're right, they're not used right now. I've basically added them with #22646 in mind, figuring we'd get them in early.

If you'd like them to show up, this is how I tested them:
```diff
--- a/src/openrct2-ui/windows/LoadSave.cpp
+++ b/src/openrct2-ui/windows/LoadSave.cpp
@@ -376,7 +376,7 @@ namespace OpenRCT2::Ui::Windows
             bool foundImage = false;
             for (auto& image : _preview.images)
             {
-                if (image.type == PreviewImageType::screenshot)
+                if (image.type == PreviewImageType::miniMap)
                 {
                     auto imagePos = frameStartPos + ScreenCoordsXY(1, 1);
                     drawPreviewImage(image, dpi, imagePos);
```